### PR TITLE
Button: use hooks

### PR DESCRIPTION
**Pull request by CloudPower97**
# Description

Part of #928 
Closes #918 as well.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Updated both `spec`s and `snapshot`s

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have not generated a new package version. (the maintainers will handle that)

### CODE REVIEW

**Comment by CloudPower97** [comment]
ping @alextrastero :smile:

File: src/Button.js [hunk 1]
```diff
@@ -1,122 +1,109 @@
...
+  fab,
+  modal,
+  flat,
+  floating,
+  large,
+  small,
+  disabled,
+  waves,
+  tooltip,
+  tooltipOptions = {},
```
Comment (alextrastero) [inline src/Button.js]: can we move to default props since we have it defined?

File: src/Button.js [hunk 2]
```diff
@@ -1,122 +1,109 @@
...
-        ref={el => (this._floatingActionBtn = el)}
+        {...props}
+        ref={_fab.current}
         className={cx('fixed-action-btn', {
-          toolbar: fab && fab.toolbarEnabled
+          toolbar: fab.toolbarEnabled
         })}
       >
-        <a className={classes}>{icon}</a>
+        <a className={classes}>{icon || <Icon>edit</Icon>}</a>
```
Comment (alextrastero) [inline src/Button.js]: let's keep `icon` as a node prop?

**Comment by alextrastero** [comment]
Perfect, thx!

**Pull request merged**